js/repl/compile.js

```diff
@@ -69,15 +96,30 @@ export default function compile(code: string, config: CompileConfig): Return {
   }
 
   try {
+    const start = Date.now();
     const transformed = Babel.transform(code, {
       babelrc: false,
       filename: "repl",
       presets: config.presets,
       sourceMap: config.sourceMap,
+      parserOpts: {
+        tokens: config.astTokens,
+      },
     });
 
+    transformTime = Date.now() - start;
+
     compiled = transformed.code;
 
+    if (!config.astLocations) {
+      transformed.ast = filterLocation(transformed.ast);
+    }
+    if (!config.astTokens) {
+      delete transformed.ast.tokens;
+    }
+
+    ast = JSON.stringify(transformed.ast, null, 2);
```

Comment:
I'm just wondering, the AST shows the emitted code, wouldn't it more practical to show the input?

For example JSX: [REPL](https://deploy-preview-1421--babel.netlify.com/repl/#?astLocations=false&astTokens=false&babili=false&browsers=&build=&builtIns=false&code_lz=DwYQ9gtgDmB2CmsAuACA9APiA&debug=false&circleciRepo=&evaluate=false&lineWrap=true&outputAST=true&outputCode=false&presets=es2015%2Creact%2Cstage-2&prettier=false&targets=&version=6.26.0)

Reply:
Good point, you can of course by hand disable all presets, but that might be forgotten.

So I see to things to do here:
* Either ensure that no presets and plugins are supplied to babel when outputing the AST.
* Have a separate compiler that uses babylon directly. 

The first one is probably easier. And in any case all other options (presets, env) should be disabled in the UI when selecting AST.

Reply:
I would take the second option.

You have two possible outputs:
- A Babel transform (as currently implemented)
- An AST, Babylon could directly parse it using a few Babylon plugin from the plugins/presets.